### DISCUSSION TITLE
disconnect and connect randomly

### DISCUSSION THREAD

**Discussion by rahul4tech** [socketio/socket.io#4620] [Q&A]
Hi,
I really need help, i have chat application used by 20-30 users. randomly socket disconnect and reconnect of my some user i checked networking everything but i did not find any problem with that, so thought of updating version 
from :
socket.io  ^4.4.0  →  ^4.5.4     
socket.io-client  ^4.4.1  →  ^4.5.4 

still it did not help.
I am really frustrated i am almost trying to find fix this from almost a month and still its not fixed.
i just see Reason : transport close of disconnection but my internet is very stable 

I am using : 
socket.io-client with Nextjs hosted on vercel
socket.io Express nodejs hosted on heroku

really need help.

Thank you

**Comment by darrachequesne** [comment]
Hi! Does it reconnect properly? If that's the case, then there is not much we can do, you shouldn't expect the connection to be always up.

**Reply by rahul4tech** [reply]
it reconnects after 2-3 seconds but i see frequent disconnects. at least if its connected for 2-3 hours i am satisfied but how can the socket disconnects in sometime less than 2 minutes.
Can't we prevent disconnects.
can you tell me is "forcenew" is something which can help ?
is their anything which can help.
if not i am not sure i have to look for other solution like pusher or ably, still i am not sure this will happen on them also or not ?

**Reply by darrachequesne** [reply]
Did you find a solution?

In most cases, the disconnections are related to the network/configuration of the elements that are between the client and the server, so again there is not much we can do.

The `forceNew` option is used for multiplexing, this won't impact the disconnections.

Reference: https://socket.io/docs/v4/client-options/#multiplex

See also: https://socket.io/docs/v4/troubleshooting-connection-issues/
